### PR TITLE
Move calculation of scale inverse to framework

### PR DESCRIPTION
**Pull request by ksivaman**

### CODE REVIEW

File: transformer_engine/pytorch/module.py [hunk 1]
```diff
@@ -323,7 +335,12 @@ def fp8_init(self, num_gemms: int = 1) -> None:
         # Allocate scales and amaxes
         self.init_fp8_meta_tensors()
 
-    def pre_forward(self, inp: torch.Tensor, num_gemms: int = 1) -> None:
+    def pre_forward(
```
Comment (timmoon10) [inline transformer_engine/pytorch/module.py:338]: Be advised that https://github.com/NVIDIA/TransformerEngine/pull/50 changes this function into a context manager: https://github.com/ptrendx/TransformerEngine/blob/e9d85b8437c0a1701703a4257c8201d2be83a931/transformer_engine/pytorch/module.py#L365

**Review by timmoon10 — COMMENTED** [review]
https://github.com/NVIDIA/TransformerEngine/pull/47 added some GeGLU kernels that will need to be modified. Pinging @zlsh80826. The relevant files are:
- https://github.com/NVIDIA/TransformerEngine/blob/main/transformer_engine/common/activation/gelu.cu
- https://github.com/NVIDIA/TransformerEngine/blob/main/transformer_engine/common/util/vectorized_pointwise.h
- https://github.com/NVIDIA/TransformerEngine/blob/main/transformer_engine/common/transpose/cast_transpose_fusion.cu

**Review by ptrendx — APPROVED** [review]
LGTM

**Comment by ptrendx** [comment]
Checked the status of TE CI internally.

**Pull request merged**